src/charm.py

```diff
@@ -73,13 +73,18 @@ def _on_install(self, _: ops.InstallEvent):
     def _on_start(self, event: ops.StartEvent):
         """Handle start event."""
         if not self.context.peer_relation:
-            self.unit.status = ops.WaitingStatus("waiting for peer relation")
+            self._set_status(Status.NO_PEER_RELATION)
             event.defer()
             return
 
-        if self.unit.is_leader() and not self.context.cluster.internal_user_credentials:
-            logger.info("Creating internal user")
+        if self.context.cluster.internal_user_credentials:
```

Comment:
**question** I was expecting only the leader unit to create/update admin/internal users, something like:

```
if unit is leader:
    if credentials_exists:
        self.auth_manager.update_admin_user()
    else:
        self.auth_manager.create_internal_user()
else:
    if credentials_exists:
         return 
    else:
         event.defer()
```

Reply:
Credentials are stored on a file. All units should have a copy of the credentials file. But only the leader _creates_ the `operator` user. `update_admin_user` will just look at internal databag credentials and recreate that part of the file.

Also, since you mention it, I noticed that a password rotation won't update the credentials on non-leader units, so that's a bug that need fixing.